# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1242819438 -7200
#      Wed May 20 13:37:18 2009 +0200
# Node ID 659657942ba687ddc7cfb0a7bf2b1072db35a6d1
# Parent  70177bf9cc16f228abfe140c1540f1addeece76f
declare octave_exit using a typedef

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-05-20  Jaroslav Hajek  <highegg@gmail.com>
+
+	* toplev.h (octave_exit_func): New typedef.
+	(octave_exit): Change to octave_exit_func.
+	
 2009-05-20  Jaroslav Hajek  <highegg@gmail.com>
 	
 	* ov-typeinfo.cc: Don't include oct-obj.h.
 
 2009-05-20  Jaroslav Hajek  <highegg@gmail.com>
 
 	* toplev.h (octave_quit_exception): New class.
 	(octave_exit): New global variable.
diff --git a/src/toplev.h b/src/toplev.h
--- a/src/toplev.h
+++ b/src/toplev.h
@@ -38,17 +38,18 @@ class tree_statement_list;
 class charMatrix;
 
 #include "quit.h"
 
 #include "input.h"
 #include "oct-map.h"
 
 
-extern OCTINTERP_API void (*octave_exit) (int);
+typedef void (*octave_exit_func) (int);
+extern OCTINTERP_API octave_exit_func octave_exit;
 
 // quit is a lot like an interrupt, so we subclass it to simplify possible
 // handling.
 class octave_quit_exception 
 : public octave_interrupt_exception
 {
 public:
   int status;
